crates/sui-storage/src/event_store/sql.rs

```diff
@@ -443,15 +416,37 @@ impl EventStore for SqlEventStore {
         Ok(rows_affected)
     }
 
+    #[instrument(level = "debug", skip_all, err)]
+    async fn all_events(
+        &self,
+        cursor: EventID,
+        limit: usize,
+        descending: bool,
+    ) -> Result<Vec<StoredEvent>, SuiError> {
+        let query = get_event_query(vec![], descending);
+        let rows = sqlx::query(&query)
+            .persistent(true)
+            .bind(cursor as i64)
+            .bind(limit as i64)
+            .map(StoredEvent::from)
+            .fetch_all(&self.pool)
+            .await
+            .map_err(convert_sqlx_err)?;
+        Ok(rows)
+    }
+
     #[instrument(level = "debug", skip_all, err)]
     async fn events_by_transaction(
         &self,
+        cursor: EventID,
```

Comment:
putting `cursor` before the primary parameter of the call (`digest` here) instead of with the rest of the queries-organization parameters (for lack of a better term) like limit & order feels off.

I would expect the signature for all these to put `cursor` directly before `limit`.

```db.events_by_transaction(txId, 0, 50, true)```

puts the most relevant data first, compared to
```db.events_by_transaction(0, txId, 50, true)```